Index: tmp.BqMy4aY8YI/src/lib/ScanlineTextureMapperContext.cpp
===================================================================
--- tmp.BqMy4aY8YI.orig/src/lib/ScanlineTextureMapperContext.cpp	2011-06-07 22:40:08.000000000 +0200
+++ tmp.BqMy4aY8YI/src/lib/ScanlineTextureMapperContext.cpp	2011-06-07 22:40:18.000000000 +0200
@@ -176,8 +176,8 @@
                     nextTile( posX, posY );
                     itLon = prevPixelX + m_toTileCoordinatesLon;
                     itLat = prevPixelY + m_toTileCoordinatesLat;
-                    posX = itLon + itStepLon * j;
-                    posY = itLat + itStepLat * j;
+                    posX = qMax<qreal>( 0.0, qMin<qreal>( tileWidth-1.0, itLon + itStepLon * j ) );
+                    posY = qMax<qreal>( 0.0, qMin<qreal>( tileHeight-1.0, itLat + itStepLat * j ) );
                     oldPosX = -1;
                 }
 
