Index: marble-1.1.90/data/CMakeLists.txt
===================================================================
--- marble-1.1.90.orig/data/CMakeLists.txt	2011-06-22 18:49:32.000000000 +0200
+++ marble-1.1.90/data/CMakeLists.txt	2011-06-22 18:49:49.000000000 +0200
@@ -247,6 +247,11 @@
     install (FILES placemarks/cityplacemarks.cache DESTINATION ${MARBLE_DATA_INSTALL_PATH}/placemarks)
 endif(MOBILE)
 
+FILE (GLOB LANG_FILES lang/marble-*)
+install (FILES
+  ${LANG_FILES}
+DESTINATION ${MARBLE_DATA_INSTALL_PATH}/lang)
+
 if(NOT APPLE AND NOT WIN32)
   if(QTONLY)
     set(ICON_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/../../usr/share/icons)
