Index: src-fremantle/data/CMakeLists.txt
===================================================================
--- src-fremantle.orig/data/CMakeLists.txt	2012-08-12 13:18:44.000000000 +0200
+++ src-fremantle/data/CMakeLists.txt	2012-08-12 19:34:06.000000000 +0200
@@ -377,6 +377,7 @@
 install (FILES
   ${AUDIO_FILES}
 DESTINATION ${MARBLE_DATA_INSTALL_PATH}/audio)
+install (DIRECTORY audio/speakers DESTINATION ${MARBLE_DATA_INSTALL_PATH}/audio)
 
 FILE (GLOB_RECURSE DGML_FILES maps/*.dgml)
 MARBLE_ADD_PROJECT_RESOURCES(${DGML_FILES})
