Index: marble-0.90+svn1209861/src/CMakeLists.txt
===================================================================
--- marble-0.90+svn1209861.orig/src/CMakeLists.txt	2010-10-20 07:28:18.000000000 -0400
+++ marble-0.90+svn1209861/src/CMakeLists.txt	2010-12-28 05:56:23.000000000 -0500
@@ -202,3 +202,12 @@
   ${CMAKE_CURRENT_SOURCE_DIR}/config-marble.h.cmake
   ${CMAKE_CURRENT_BINARY_DIR}/config-marble.h
 )
+
+install ( FILES marble.sh DESTINATION bin/ PERMISSIONS
+  OWNER_EXECUTE OWNER_WRITE OWNER_READ
+  GROUP_EXECUTE GROUP_READ
+  WORLD_READ WORLD_EXECUTE)
+install ( FILES marble.desktop DESTINATION
+  ${CMAKE_INSTALL_PREFIX}/../../usr/share/applications/hildon/ )
+install ( FILES marble.service DESTINATION
+  ${CMAKE_INSTALL_PREFIX}/../../usr/share/dbus-1/services/ )
Index: marble-0.90+svn1209861/src/marble.desktop
===================================================================
--- marble-0.90+svn1209861.orig/src/marble.desktop	2010-12-15 00:54:43.000000000 -0500
+++ marble-0.90+svn1209861/src/marble.desktop	2010-12-28 05:56:23.000000000 -0500
@@ -1,7 +1,6 @@
 [Desktop Entry]
 Type=Application
-TryExec=marble
-Exec=marble
+Exec=/opt/marble/bin/marble.sh
 Name=Marble
 Name[ast]=Marble
 Name[bg]=Marble
@@ -108,5 +107,9 @@
 GenericName[zh_TW]=桌面星象儀
 Icon=marble
 Terminal=false
-Categories=Qt;KDE;Education;Geography;
+StartupWMClass=marble
+X-Window-Icon=marble
 X-DocPath=marble/index.html
+X-HildonDesk-ShowInToolbar=true
+X-Osso-Service=org.kde.edu.marble
+X-Osso-Type=application/x-executable
Index: marble-0.90+svn1209861/src/marble.sh
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ marble-0.90+svn1209861/src/marble.sh	2010-12-28 05:56:45.000000000 -0500
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+# Set the environment variable XDG_DATA_HOME to have
+# marble store cached data in /home/user/MyDocs/marble/data
+# Can be overridden by specifying an alternative path in
+# the environment variable MARBLE_DATA_HOME, e.g.
+# $ export MARBLE_DATA_HOME=/home/user
+# which would result in cached data being written to
+# /home/user/marble/data.
+XDG_DATA_HOME="${MARBLE_DATA_HOME:-${HOME}/MyDocs/.local/share}"
+export XDG_DATA_HOME
+
+# Marble is installed in /opt/marble. Add it's library
+# to the library search path.
+LD_LIBRARY_PATH="/opt/marble/lib:${LD_LIBRARY_PATH}"
+export LD_LIBRARY_PATH
+
+PATH="/opt/marble/bin:${PATH}"
+export PATH
+
+/opt/marble/bin/marble $*
Index: marble-0.90+svn1209861/src/marble.service
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ marble-0.90+svn1209861/src/marble.service	2010-12-28 05:56:23.000000000 -0500
@@ -0,0 +1,3 @@
+[D-BUS Service]
+Name=org.kde.edu.marble
+Exec=/opt/marble/bin/marble.sh
Index: marble-0.90+svn1209861/data/CMakeLists.txt
===================================================================
--- marble-0.90+svn1209861.orig/data/CMakeLists.txt	2010-11-28 14:58:04.000000000 -0500
+++ marble-0.90+svn1209861/data/CMakeLists.txt	2010-12-28 05:56:23.000000000 -0500
@@ -231,11 +231,12 @@
 
 if(NOT APPLE AND NOT WIN32)
   if(QTONLY)
-    set(ICON_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/icons)
+    set(ICON_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/../../usr/share/icons)
   endif(QTONLY)
     install (FILES icons/hi128-app-marble.png DESTINATION ${ICON_INSTALL_DIR}/hicolor/128x128/apps/ RENAME marble.png)
     install (FILES icons/hi64-app-marble.png DESTINATION ${ICON_INSTALL_DIR}/hicolor/64x64/apps/ RENAME marble.png)
     install (FILES icons/hi48-app-marble.png DESTINATION ${ICON_INSTALL_DIR}/hicolor/48x48/apps/ RENAME marble.png)
+    install (FILES icons/hi32-app-marble.png DESTINATION ${ICON_INSTALL_DIR}/hicolor/26x26/apps/ RENAME marble.png)
     install (FILES icons/hi32-app-marble.png DESTINATION ${ICON_INSTALL_DIR}/hicolor/32x32/apps/ RENAME marble.png)
     install (FILES icons/hi22-app-marble.png DESTINATION ${ICON_INSTALL_DIR}/hicolor/22x22/apps/ RENAME marble.png)
     install (FILES icons/hi16-app-marble.png DESTINATION ${ICON_INSTALL_DIR}/hicolor/16x16/apps/ RENAME marble.png)
