Index: src-fremantle/data/CMakeLists.txt
===================================================================
--- src-fremantle.orig/data/CMakeLists.txt	2012-08-12 12:54:15.000000000 +0200
+++ src-fremantle/data/CMakeLists.txt	2012-08-12 12:54:20.000000000 +0200
@@ -141,6 +141,62 @@
 DESTINATION ${MARBLE_DATA_INSTALL_PATH}/maps/earth/hillshading/0/0)
 
 install (FILES
+maps/earth/public-transport/public-transport.png
+maps/earth/public-transport/legend.html
+maps/earth/public-transport/public-transport.dgml
+DESTINATION ${MARBLE_DATA_INSTALL_PATH}/maps/earth/public-transport)
+
+install (FILES
+maps/earth/public-transport/legend/train_station.png
+maps/earth/public-transport/legend/subway_station.png
+maps/earth/public-transport/legend/light_rail_station.png
+maps/earth/public-transport/legend/bus_stop.png
+maps/earth/public-transport/legend/tram_station.png
+DESTINATION ${MARBLE_DATA_INSTALL_PATH}/maps/earth/public-transport/legend)
+
+install (FILES
+maps/earth/public-transport/0/0/0.png
+DESTINATION ${MARBLE_DATA_INSTALL_PATH}/maps/earth/public-transport/0/0)
+
+install (FILES
+maps/earth/hikebikemap/hikebikemap.dgml
+maps/earth/hikebikemap/hikebikemap-preview.png
+maps/earth/hikebikemap/legend.html
+DESTINATION ${MARBLE_DATA_INSTALL_PATH}/maps/earth/hikebikemap)
+
+install (FILES
+maps/earth/hikebikemap/legend/access_no.png
+maps/earth/hikebikemap/legend/bridleway.png
+maps/earth/hikebikemap/legend/cycleway.png
+maps/earth/hikebikemap/legend/farmland.png
+maps/earth/hikebikemap/legend/footway.png
+maps/earth/hikebikemap/legend/meadow_stream.png
+maps/earth/hikebikemap/legend/motorway.png
+maps/earth/hikebikemap/legend/parking_building.png
+maps/earth/hikebikemap/legend/path.png
+maps/earth/hikebikemap/legend/power_tower_line.png
+maps/earth/hikebikemap/legend/primary.png
+maps/earth/hikebikemap/legend/rail.png
+maps/earth/hikebikemap/legend/residential.png
+maps/earth/hikebikemap/legend/secondary.png
+maps/earth/hikebikemap/legend/tertiary.png
+maps/earth/hikebikemap/legend/track1.png
+maps/earth/hikebikemap/legend/track2.png
+maps/earth/hikebikemap/legend/track3.png
+maps/earth/hikebikemap/legend/track4.png
+maps/earth/hikebikemap/legend/track5.png
+maps/earth/hikebikemap/legend/water_allotments.png
+DESTINATION ${MARBLE_DATA_INSTALL_PATH}/maps/earth/hikebikemap/legend)
+
+install (FILES
+maps/earth/hikebikemap/0/0/0.png
+DESTINATION ${MARBLE_DATA_INSTALL_PATH}/maps/earth/hikebikemap/0/0)
+
+install (FILES
+maps/earth/hikebikemap/lighting/0/0/0.png
+DESTINATION ${MARBLE_DATA_INSTALL_PATH}/maps/earth/hikebikemap/lighting/0/0)
+
+install (FILES
 maps/earth/temp-july/temp-july.jpg
 maps/earth/temp-july/temp-july-preview.png
 maps/earth/temp-july/temp-july.dgml
